packages/nextjs/app/builders/[address]/_components/UpdateLocationModal.tsx

```diff
@@ -0,0 +1,84 @@
+import { forwardRef, useEffect, useState } from "react";
+import ReactFlagsSelect from "react-flags-select";
+import { useUpdateLocation } from "~~/hooks/useUpdateLocation";
+import { UserLocation } from "~~/services/database/repositories/users";
+
+type UpdateLocationModalProps = {
+  closeModal: () => void;
+  existingLocation: UserLocation;
+  isOpen: boolean;
+};
+
+export const UpdateLocationModal = forwardRef<HTMLDivElement, UpdateLocationModalProps>(
+  ({ closeModal, existingLocation, isOpen }, ref) => {
+    const [selectedCountry, setSelectedCountry] = useState(existingLocation || "");
+    const { updateLocation, isPending } = useUpdateLocation({
+      onSuccess: closeModal,
+    });
+
+    useEffect(() => {
+      if (!isOpen) return;
+      const handleKeyDown = (e: KeyboardEvent) => {
+        if (e.key === "Escape") {
+          closeModal();
+        }
+      };
+      window.addEventListener("keydown", handleKeyDown);
+      return () => {
+        window.removeEventListener("keydown", handleKeyDown);
+      };
+    }, [isOpen, closeModal]);
```

Comment:
Not using `daisyUI` modal here since the using it, it uses `dialo` element which causes the `react-flags-select` render very weirdly and hard to style since dialog is put on `top-layer` and having z-index or anything to ReactFlags selsect was not working so had to implement own modal. 

Tried keeping it very simillar to daisyUI modal